pkg/scalers/aws_sigv4.go

```diff
@@ -0,0 +1,122 @@
+package scalers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+// SigV4Config configures signing requests with SigV4.
+type SigV4Config struct {
+	Enabled bool   `yaml:"enabled,omitempty"`
+	Region  string `yaml:"region,omitempty"`
+}
+
+type awsAMPMetadata struct {
+	awsRegion        string
+	awsAuthorization awsAuthorizationMetadata
+}
+
+// Custom round tripper to sign requests
+type roundTripper struct {
+	signer *v4.Signer
+	region string
+}
+
+var (
+	// ErrAwsAMPNoAwsRegion is returned when "awsRegion" is missing from the config.
+	ErrAwsAMPNoAwsRegion = errors.New("no awsRegion given")
+)
+
+func (rt *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+
+	// Sign request
+	rt.signer.Sign(req, nil, "aps", rt.region, time.Now())
+
+	// Create default transport
+	transport := &http.Transport{}
+
+	// Send signed request
+	return transport.RoundTrip(req)
+}
+
+// NewSigV4RoundTripper returns a new http.RoundTripper that will sign requests
+// using Amazon's Signature Verification V4 signing procedure. The request will
+// then be handed off to the next RoundTripper provided by next. If next is nil,
+// http.DefaultTransport will be used.
+//
+// Credentials for signing are retrieving used the default AWS credential chain.
+// If credentials could not be found, an error will be returned.
+func NewSigV4RoundTripper(config *ScalerConfig, next http.RoundTripper) (http.RoundTripper, error) {
+	cfg := aws.Config{}
+
+	triggerMetadata := config.TriggerMetadata
+	if triggerMetadata == nil {
+		return nil, fmt.Errorf("trigger metadata cannot be nil")
+	}
+
+	awsRegion := triggerMetadata["awsRegion"]
+	if awsRegion == "" {
+		return nil, fmt.Errorf("awsRegion not configured in trigger metadata")
+	}
+
+	metadata, err := parseAwsAMPMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing AMP metadata: %w", err)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := session.NewSession(&cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get a new aws session")
+	}
+
+	if *session.Config.Region == "" {
+		awsRegion = "us-east-1"
+	}
+
+	var creds *credentials.Credentials
+	if metadata.awsAuthorization.awsAccessKeyID != "" && metadata.awsAuthorization.awsSecretAccessKey != "" {
+		creds = credentials.NewStaticCredentials(metadata.awsAuthorization.awsAccessKeyID, metadata.awsAuthorization.awsSecretAccessKey, "")
+		//session.Config.Credentials = creds
+	} else {
+		creds = session.Config.Credentials
+	}
```

Comment:
Don't generate the credentials here, please use [`getAwsConfig`](https://github.com/kedacore/keda/blob/main/pkg/scalers/aws_common.go#L33).
That function wraps the management of pod identities and/or credentials

Reply:
incorporated the changes